src/settings/mod.rs

```diff
@@ -13,26 +14,25 @@ use runtime_settings::RuntimeSettings;
 
 #[derive(Clone, Debug, Tree)]
 pub struct Settings {
-    #[tree(depth = 4)]
     pub booster: RuntimeSettings,
 
     #[tree(skip)]
     pub mac: smoltcp_nal::smoltcp::wire::EthernetAddress,
 
-    pub ip: Cidr,
-    pub broker: String<255>,
-    pub gateway: IpAddr,
-    pub id: String<23>,
+    pub ip: Leaf<Cidr>,
+    pub broker: StrLeaf<String<255>>,
+    pub gateway: Leaf<IpAddr>,
+    pub id: StrLeaf<String<23>>,
 }
 
-impl serial_settings::Settings<5> for Settings {
+impl serial_settings::Settings for Settings {
     fn reset(&mut self) {
         self.id.clear();
         write!(&mut self.id, "{}", self.mac).unwrap();
 
         self.booster.reset();
-        self.ip = "0.0.0.0/0".parse().unwrap();
-        self.broker = "mqtt".parse().unwrap();
-        self.gateway = "0.0.0.0".parse().unwrap();
+        self.ip = Leaf::from(Cidr::from_str("0.0.0.0/0").unwrap());
```

Comment:
Why the change from `parse()`?

Reply:
Because `.parse::<String<_>>()` apparently doesn't work (yet), and I didn't want to explicitly specify the size:

```
const arguments cannot yet be inferred with `_`
```

Reply:
Okay, with only `Leaf` it resolves the type using `parse` again.